In [1]:
#import libraries
import pandas as pd

In [2]:
#load in answers dataset
a = pd.read_csv('Answers.csv', usecols = ['Id', 'OwnerUserId', 'CreationDate', 'ParentId', 'Score', 'Body'], low_memory=False)
print(a.shape)
a.head()

(1048575, 6)


Id  OwnerUserId          CreationDate  ParentId  Score  \
0   92         61.0  2008-08-01T14:45:37Z        90     13   
1  124         26.0  2008-08-01T16:09:47Z        80     12   
2  199         50.0  2008-08-01T19:36:46Z       180      1   
3  269         91.0  2008-08-01T23:49:57Z       260      4   
4  307         49.0  2008-08-02T01:49:46Z       260     28   

                                                Body  
0  <p><a href="http://svnbook.red-bean.com/">Vers...  
1  <p>I wound up using this. It is a kind of a ha...  
2  <p>I've read somewhere the human eye can't dis...  
3  <p>Yes, I thought about that, but I soon figur...  
4  <p><a href="http://www.codeproject.com/Article...

In [3]:
#load in questions dataset
q = pd.read_csv('Questions.csv', usecols = ['Id', 'OwnerUserId', 'CreationDate', 'ClosedDate', 'Score', 'Title', 'Body'], low_memory=False)
print(q.shape)
q.head()

(1048575, 7)


Id OwnerUserId          CreationDate            ClosedDate Score  \
0   80          26  2008-08-01T13:57:07Z                   NaN    26   
1   90          58  2008-08-01T14:41:24Z  2012-12-26T03:45:49Z   144   
2  120          83  2008-08-01T15:50:08Z                   NaN    21   
3  180     2089740  2008-08-01T18:42:19Z                   NaN    53   
4  260          91  2008-08-01T23:22:08Z                   NaN    49   

                                               Title  \
0  SQLStatement.execute() - multiple queries in o...   
1  Good branching and merging tutorials for Torto...   
2                                  ASP.NET Site Maps   
3                 Function for creating color wheels   
4  Adding scripting functionality to .NET applica...   

                                                Body  
0  <p>I've written a database generation script i...  
1  <p>Are there any really good tutorials explain...  
2  <p>Has anyone got experience creating <strong>...  
3  <p>This is something I've pseudo-solved many t...  
4  <p>I have a little game written in C#. It uses...

In [4]:
#load in tags dataset
t = pd.read_csv('Tags.csv', usecols = ['Id', 'Tag'], low_memory=False)
print(t.shape)
t.head()

(1048575, 2)


Id             Tag
0  80            flex
1  80  actionscript-3
2  80             air
3  90             svn
4  90     tortoisesvn

In [17]:
tags = t['Tag'].value_counts(ascending=False).reset_index()
tags.head(15)

Tag  count
0            c#  34979
1          java  29266
2           php  27837
3    javascript  26199
4        jquery  22814
5       android  22198
6           c++  14691
7        iphone  14482
8        python  13470
9       asp.net  13433
10         .net  12802
11         html  12402
12        mysql  11863
13  objective-c  10128
14          sql   9742

We can see that the languages we are interested in (java, python, c++) are all in the 15 most used tags within the dataset. We will now use the tags to remove any data we aren't interested in. 

Using the count of these tags, we know our cleaned tag dataset should have (29266+14691+13470) 57427 entries

In [21]:
t_java = t[t.Tag == 'java']
print("Java: ", t_java.shape)

t_python = t[t.Tag == 'python']
print("Python: ", t_python.shape)

t_cpp = t[t.Tag == 'c++']
print("C++: ", t_cpp.shape)

Java:  (29266, 2)
Python:  (13470, 2)
C++:  (14691, 2)


Now we have the Ids for the three relevent tags in three separate datasets. We will use these to scrape the relevent rows from the questions and answers datasets. 